pkg/dns/dns.go

```diff
@@ -49,101 +40,28 @@ const (
 )
 
 type DNSResolver struct {
-	DnsResolverChan   chan []*clusterv3.Cluster
 	client            *dns.Client
-	resolvConfServers []string
-	cache             map[string]*domainCacheEntry
-	// adsCache is used for update bpf map
-	adsCache *ads.AdsCache
-	// dns refresh priority queue based on exp
-	dnsRefreshQueue workqueue.TypedDelayingInterface[any]
+	ResolvConfServers []string
+	Cache             map[string]*DomainCacheEntry
 	sync.RWMutex
 }
 
-type domainCacheEntry struct {
+type DomainCacheEntry struct {
 	addresses []string
 }
 
-// pending resolve domain info,
+// pending resolve domain info of Kennel-Native Mode,
 // domain name is used for dns resolution
 // cluster is used for create the apicluster
-type pendingResolveDomain struct {
-	domainName  string
-	clusters    []*clusterv3.Cluster
-	refreshRate time.Duration
-}
-
-func overwriteDnsCluster(cluster *clusterv3.Cluster, domain string, addrs []string) bool {
-	buildLbEndpoints := func(port uint32) []*endpointv3.LbEndpoint {
-		lbEndpoints := make([]*endpointv3.LbEndpoint, 0, len(addrs))
-		for _, addr := range addrs {
-			ip := net.ParseIP(addr)
-			if ip == nil {
-				continue
-			}
-			if ip.To4() == nil {
-				continue
-			}
-			lbEndpoint := &endpointv3.LbEndpoint{
-				HealthStatus: v3.HealthStatus_HEALTHY,
-				HostIdentifier: &endpointv3.LbEndpoint_Endpoint{
-					Endpoint: &endpointv3.Endpoint{
-						Address: &v3.Address{
-							Address: &v3.Address_SocketAddress{
-								SocketAddress: &v3.SocketAddress{
-									Address: addr,
-									PortSpecifier: &v3.SocketAddress_PortValue{
-										PortValue: port,
-									},
-								},
-							},
-						},
-					},
-				},
-				// TODO: support LoadBalancingWeight
-				LoadBalancingWeight: &wrapperspb.UInt32Value{
-					Value: 1,
-				},
-			}
-			lbEndpoints = append(lbEndpoints, lbEndpoint)
-		}
-		return lbEndpoints
-	}
-
-	ready := true
-	for _, e := range cluster.LoadAssignment.Endpoints {
-		pos := -1
-		var lbEndpoints []*endpointv3.LbEndpoint
-		for i, le := range e.LbEndpoints {
-			socketAddr, ok := le.GetEndpoint().GetAddress().GetAddress().(*v3.Address_SocketAddress)
-			if !ok {
-				continue
-			}
-			_, err := netip.ParseAddr(socketAddr.SocketAddress.Address)
-			if err != nil {
-				if socketAddr.SocketAddress.Address == domain {
-					pos = i
-					lbEndpoints = buildLbEndpoints(socketAddr.SocketAddress.GetPortValue())
-				} else {
-					// There is other domains not resolved for this cluster
-					ready = false
-				}
-			}
-		}
-		if pos >= 0 {
-			e.LbEndpoints = slices.Replace(e.LbEndpoints, pos, pos+1, lbEndpoints...)
-		}
-	}
-
-	return ready
+type PendingResolveDomain struct {
```

Comment:
How do you plean to support dual-engine then? 
If you are going to add another struct here, wouldn't that still coupling